server/service/schedule/schedule.go

```diff
@@ -282,26 +378,26 @@ func (s *Schedule) Done() <-chan struct{} {
 	return s.done
 }
 
+// getScheduleInterval returns the schedule interval after truncating the duration to seconds and
+// applying a one second floor (e.g., 600ms becomes 1s, 1300ms becomes 2s, 1000ms becomes 2s)
 func (s *Schedule) getSchedInterval() time.Duration {
 	s.schedIntervalMu.Lock()
 	defer s.schedIntervalMu.Unlock()
 
-	return s.schedInterval
+	return truncateSecondsWithFloor(s.schedInterval)
```

Comment:
I think you should decide whether you want to truncate when you return, or when you set `s.schedInterval`, both feel like an overkill.